### PR TITLE
Delete assured services

### LINKED ISSUE #88: Make Service Assurance Types less MN-specific
The service assurance stuff seems super specific to MN.  At some point we need to figure out whether that is generalizable structure or should be removed entirely.

### PR DESCRIPTION
**Pull request by jasonaowen**
Once upon a time, there was a program in Minnesota for helping people deal with chemical dependencies (ie, addiction) (there [still is](http://www.dhs.state.mn.us/main/idcplg?IdcService=GET_DYNAMIC_CONVERSION&RevisionSelectionMethod=LatestReleased&dDocName=id_054366), but [there once was, too](http://weknowmemes.com/wp-content/uploads/2011/10/i-used-to-be-socially-awkward-mitch-hedberg.jpg)). Providers who took part in this program sometimes needed to bill Medicaid for their services, and so the Provider Screening Module started to be extended to allow screening such providers.

This seems to have been designed as a set of assured services that providers provided. `ProviderProfile`s were extended to include those assured services, and UI was designed to collect statements from providers relating to those services.

Then, one dark day, Minnesota's Provider Screening Module was shelved, undeployed, and the work that had been done to date to support the chemical dependency program was left unfinished.

Now, we lay this code to rest, gone but not forgotten.

---

This is a larger pull request, but it's almost entirely deletions, and I've split it up into small steps that made sense to me as incremental deletions - hopefully it also makes sense to you. I suggest reviewing commit-by-commit.

The rough outline is:
- a [whitespace-only](https://github.com/OpenTechStrategies/psm/commit/bf18d6527498efdce7638f4e88e06ddc9d87a5c5?w=1) commit
- removing the unreferenced UI
- removing the code backing that UI
- removing the types that only that code referenced

Testing this is either really easy or really hard: there should be no change in behavior, since the feature was never enabled. I was able to build, deploy, submit enrollments, review them, reject them, and approve them.

This does involve database schema changes, so I suggest running [`drop owned by current_user;`](https://www.postgresql.org/docs/current/static/sql-drop-owned.html) in a `psql` session, then re-importing the schema with `cat db/{legacy_seed,jbpm,seed}.sql | psql -h localhost -U psm psm`.

---

Related pull requests:
- #72 Convert ServiceAssuranceType to annotation
- #93 Redo ServiceAssuranceExtType as annotations

Issue #36 Use Hibernate 5, instead of 4
Resolves #88 Make Service Assurance Types less MN-specific

### CODE REVIEW

**Comment by jasonaowen** [comment]
Rebased onto master to resolve merge conflicts.

**Comment by jasonaowen** [comment]
Rebased onto master to resolve merge conflicts.

File: psm-app/cms-business-process/src/main/resources/META-INF/Medicaid.hbm.xml [hunk 1]
```diff
@@ -2,38 +2,6 @@
 <!DOCTYPE hibernate-mapping PUBLIC "-//Hibernate/Hibernate Mapping DTD 3.0//EN"
                                    "http://hibernate.sourceforge.net/hibernate-mapping-3.0.dtd">
 <hibernate-mapping default-lazy="false">
-	<class name="gov.medicaid.entities.AssuredService" table="PROFILE_ASSURED_SVC_XREF">
```
Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/resources/META-INF/Medicaid.hbm.xml]: So this one entity was associated with two tables?

**Comment by brainwane** [comment]
I tested a few basic bits of functionality with this branch just now. A system administrator can create a new provider user, a provider can submit an enrollment, and a service admin can approve that enrollment. 👍

File: psm-app/cms-business-process/src/main/resources/META-INF/Medicaid.hbm.xml (hunk 1 above)
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/resources/META-INF/Medicaid.hbm.xml]: Right! See [line 18](https://github.com/OpenTechStrategies/psm/pull/239/files/46a4d51551a82ea3321b6eeab808d2ec9924546c..5d83880550d99652de25d6a82a0efd383a4a8b21#diff-26153ff6bb5ab0b260a281a24041613bL18).

**Comment by BenGalewsky** [comment]
Thanks @jasonaowen  this is a very through change and doesn't seem to break anything we are able to keep working.

**Pull request merged**